src/ui/libs/DatalensChartkit/ChartKit/components/ChartKitTooltip/ChartKitTooltip.tsx

```diff
@@ -0,0 +1,135 @@
+    const anchorsRef = React.useRef(anchors);
+
+    const updateAnchors = React.useCallback((nextAnchors: TooltipAnchors) => {
+        anchorsRef.current = nextAnchors;
+        setAnchors(nextAnchors);
+    }, []);
+
+    const handleAnchorMouseEnter = React.useCallback(
+        (e: MouseEvent) => {
+            const nodeId = getNodeIdFromMouseEvent(e);
+
+            if (!anchorsRef.current[nodeId]) {
+                return;
+            }
+
+            const updatedAnchors = {...anchorsRef.current};
+            Object.entries(updatedAnchors).forEach(([id, anchor]) => {
+                anchor.open = id === nodeId;
+            });
+
+            updateAnchors(updatedAnchors);
+        },
+        [updateAnchors],
+    );
+
+    const handleAnchorMouseLeave = React.useCallback(
+        (e: MouseEvent) => {
+            const nodeId = getNodeIdFromMouseEvent(e);
+
+            if (!anchorsRef.current[nodeId]) {
+                return;
+            }
+
+            const updatedAnchors = {...anchorsRef.current};
+            updatedAnchors[nodeId].open = false;
+            updateAnchors(updatedAnchors);
+        },
+        [updateAnchors],
+    );
+
+    React.useImperativeHandle(
+        ref,
+        () => ({
+            checkForTooltipNodes(container) {
+                if (!container) {
+                    return;
+                }
+
+                const nodes = Array.from(
+                    container.querySelectorAll(`[${ATTR_DATA_TOOLTIP_CONTENT}]`),
+                ) as HTMLElement[];
+                const nextAnchors = nodes.reduce<TooltipAnchors>((acc, node) => {
+                    const id = node.id;
+                    const content = getTooltipContent(node.dataset['tooltipContent']);
+                    if (id && content) {
+                        acc[id] = {
+                            ref: {current: node},
+                            open: false,
+                            content,
+                        };
+                        node.addEventListener('mouseenter', handleAnchorMouseEnter);
+                        node.addEventListener('mouseleave', handleAnchorMouseLeave);
+                    }
+                    return acc;
+                }, {});
+                anchorsRef.current = nextAnchors;
+                setAnchors(nextAnchors);
+            },
+        }),
+        [handleAnchorMouseEnter, handleAnchorMouseLeave],
+    );
+
+    React.useEffect(() => {
+        // force Popover to recalculate its position after changing the content
+        window.dispatchEvent(new CustomEvent('scroll'));
+    });
+
+    if (!Object.keys(anchors).length) {
+        return null;
+    }
+
+    return (
+        <React.Fragment>
+            {Object.entries(anchors).map(([id, anchor]) => {
+                return (
+                    <Popup key={id} anchorRef={anchor.ref} open={anchor.open}>
+                        <div dangerouslySetInnerHTML={{__html: anchor.content}} />
```

Comment:
Maybe it's worth adding styles for the container? I would like to limit the maximum width and add padding so that a сonsistent style with the service is maintained.